MeetingMinder.agent.nut

```
@@ -0,0 +1,169 @@
+        
+        _request_url = format("https://www.googleapis.com/calendar/v3/calendars/%s/events?"
+        +"key=%s&access_token=", calendarId, API_key);
+        _clientId = clientId;
+        _tzOffset = offset;
+        _clientSecret = clientSecret;
+        _tb = server.load();
+        if("refreshToken" in _tb) { // don't need to check if accessToken is there too
+            _refreshToken = _tb.refreshToken;
+            _accessToken = _tb.accessToken;
+        }
+        else {
+            acquireTokens();
+        }
+    }
+    
+    function acquireTokens() {
+        local headers = {
+            "Content-Type" : "application/x-www-form-urlencoded",
+            "Host" : "accounts.google.com"
+        };
+        local parameters = http.urlencode({
+            "client_id" : clientId,
+            "scope" : calendarScope
+        });
+        local url = "https://accounts.google.com/o/oauth2/device/code";
+        
+        local request = http.post(url, headers, parameters);
+        local response = request.sendsync();
+        if(response.statuscode == 200) {
+            local body = http.jsondecode(response.body);
+            server.log(format("Please enter the code %s at the the link: %s", body.user_code,
+            "https://www.google.com/device"));
+            headers = {
+                "Content-Type" : "application/x-www-form-urlencoded",
+                "Host" : "www.googleapis.com"
+            };
+            parameters = http.urlencode({
+                "client_id" : _clientId,
+                "client_secret" : _clientSecret,
+                "code" : body.device_code,
+                "grant_type" : "http://oauth.net/grant_type/device/1.0"
+            });
+            url = "https://www.googleapis.com/oauth2/v4/token";
+            request = http.post(url, headers, parameters);
+            response = request.sendsync();
+            while(response.statuscode == 400) {
```

This creates a loop that the user can't interrupt or handle if things are not going well.  Would be better to bubble up an error to the user and let them handle the error.